alterconfigs.go

```diff
@@ -0,0 +1,202 @@
+	ConfigName  string
+	ConfigValue string
+}
+
+func (t alterConfigsResponseV0ConfigEntry) size() int32 {
+	return sizeofString(t.ConfigName) +
+		sizeofString(t.ConfigValue)
+}
+
+func (t alterConfigsResponseV0ConfigEntry) writeTo(wb *writeBuffer) {
+	wb.writeString(t.ConfigName)
+	wb.writeString(t.ConfigValue)
+}
+
+func (t *alterConfigsResponseV0ConfigEntry) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readString(r, size, &t.ConfigName); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ConfigValue); err != nil {
+		return
+	}
+	return
+}
+
+type alterConfigsResponseV0Resource struct {
+	ErrorCode    int16
+	ErrorMessage string
+	ResourceType int8
+	ResourceName string
+}
+
+func (t alterConfigsResponseV0Resource) size() int32 {
+	return sizeofInt16(t.ErrorCode) +
+		sizeofString(t.ErrorMessage) +
+		sizeofInt8(t.ResourceType) +
+		sizeofString(t.ResourceName)
+}
+
+func (t alterConfigsResponseV0Resource) writeTo(wb *writeBuffer) {
+	wb.writeInt16(t.ErrorCode)
+	wb.writeString(t.ErrorMessage)
+	wb.writeInt8(t.ResourceType)
+	wb.writeString(t.ResourceName)
+}
+
+func (t *alterConfigsResponseV0Resource) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt16(r, size, &t.ErrorCode); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ErrorMessage); err != nil {
+		return
+	}
+	if remain, err = readInt8(r, remain, &t.ResourceType); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ResourceName); err != nil {
+		return
+	}
+
+	return
+}
+
+//alterConfigsResponseV0 descirbe configs response V0 version
+type alterConfigsResponseV0 struct {
+	ThrottleTimeMs int32
+	Resources      []alterConfigsResponseV0Resource
+}
+
+func (t alterConfigsResponseV0) size() int32 {
+	return sizeofInt32(t.ThrottleTimeMs) +
+		sizeofArray(len(t.Resources), func(i int) int32 { return t.Resources[i].size() })
+}
+
+func (t alterConfigsResponseV0) writeTo(wb *writeBuffer) {
+	wb.writeInt32(t.ThrottleTimeMs)
+	wb.writeArray(len(t.Resources), func(i int) { t.Resources[i].writeTo(wb) })
+}
+
+func (t *alterConfigsResponseV0) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt32(r, size, &t.ThrottleTimeMs); err != nil {
+		return
+	}
+	fn := func(r *bufio.Reader, size int) (fnRemain int, fnErr error) {
+		var resource alterConfigsResponseV0Resource
+		if fnRemain, fnErr = (&resource).readFrom(r, size); err != nil {
+			return
+		}
+		t.Resources = append(t.Resources, resource)
+		return
+	}
+	if remain, err = readArrayWith(r, remain, fn); err != nil {
+		return
+	}
+
+	return
+}
+
+type Resource struct {
+	ResourceType  int8
+	ResourceName  string
+	ConfigEntries []ConfigEntry
+}
+
+//AlterConfig
+type AlterConfig struct {
+	Resources    []Resource
+	ValidateOnly bool
+}
+
+func (c *Conn) alterConfigs(request alterConfigsRequestV0) (*alterConfigsResponseV0, error) {
```

Comment:
```suggestion
func (c *Conn) alterConfigs(request alterConfigsRequestV0) (alterConfigsResponseV0, error) {
```

Here we can avoid an allocation by returning a value instead of a pointer.

Reply:
Done